File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1124.c
#include <stdio.h>

int main(){
    
    int l, c, ra, rb, xa, ya, xb, yb, cxa, cya, cxb, cyb;
    
    while(1){
        
        scanf("%d %d %d %d", &l, &c, &ra, &rb);
        
        if(l == 0 && c == 0 && ra == 0 && rb == 0) break;

        cxa = ra;
        cya = c-ra;
        
        cxb = l-rb;
        cyb = rb;
        
        if(ra*2 <= l && ra*2 <= c && rb*2 <= l && rb*2 <= c){
            if((cxa - cxb) * (cxa - cxb) + (cya - cyb) * (cya - cyb)  >= (ra + rb) * (ra + rb)) printf("S\n");
            else printf("N\n");
        } else printf("N\n");
        
    }
    
}

File: codesLeftInWindows/CP-Problems-tilnoeneBackup/URI/1124.rb
while linha = gets
    ls = linha.split(' ')

    l = (ls[0]).to_i
    c = (ls[1]).to_i
    ra= (ls[2]).to_i
    rb = (ls[3]).to_i
        
    if(l == 0 and c == 0 and ra == 0 and rb == 0) then break end
            
    cxa = ra;
    cya = c-ra;
            
    cxb = l-rb;
    cyb = rb;

    if(ra*2 <= l and ra*2 <= c and rb*2 <= l and rb*2 <= c) then
        if((cxa - cxb) * (cxa - cxb) + (cya - cyb) * (cya - cyb)  >= (ra + rb) * (ra + rb)) then puts("S")
        else puts("N") end
    else puts("N") end
end